api/v1/intrusiondetection_types.go

```diff
@@ -36,6 +36,46 @@ type IntrusionDetectionSpec struct {
 	// IntrusionDetectionControllerDeployment configures the IntrusionDetection Controller Deployment.
 	// +optional
 	IntrusionDetectionControllerDeployment *IntrusionDetectionControllerDeployment `json:"intrusionDetectionControllerDeployment,omitempty"`
+
+	// DeepPacketInspectionDaemonset configures the DPI Daemonset
+	// +optional
+	DeepPacketInspectionDaemonset *DeepPacketInspectionDaemonset `json:"deepPacketInspectionDaemonset,omitempty"`
+}
+
+type DeepPacketInspectionDaemonset struct {
+	// DPIDaemonsetSpec configures the DPI Daemonset
+	// +optional
+	Spec *DPIDaemonsetSpec `json:"spec,omitempty"`
+}
+
+type DPIDaemonsetSpec struct {
+	// Template specifies DPI Daemonset Template
+	// +optional
+	Template *DPIDaemonsetTemplate `json:"template,omitempty"`
+}
+
+type DPIDaemonsetTemplate struct {
+	// Spec specifies DPI Daemonset Template Spec
+	// +optional
+	Spec *DPIDaemonsetTemplateSpec `json:"spec,omitempty"`
+}
+
+type DPIDaemonsetTemplateSpec struct {
+	// List of DPI Daemonset Init containers definitions
+	// +kubebuilder:validation:MaxItems=1
+	InitContainers []DPIDaemonsetInitContainer `json:"initContainers,omitempty"`
+}
+
+type DPIDaemonsetInitContainer struct {
+	// Container name
```

Comment:
We should have validation on this field - we only support a single container here right now (the snort rules init container), so we should reject any thing that doesn't match a pre-established conventional name (e.g., "snort-rules").

For example: 
https://github.com/tigera/operator/blob/dcb21dfb148f47b5dfd39bd136398e04c4c21935/api/v1/calico_node_types.go#L28-L29

Reply:
Done.